dcc-portal-ui/app/scripts/analysis/js/analysisLauncher.js

```diff
@@ -133,6 +169,32 @@
     _this.launchSurvival = function(setIds) {
       return _launchAnalysis(setIds, 'survival', 'analysis/view/survival/');
     };
+    
+    _this.launchOncogridAnalysis = function (setIds) {
+      console.log('Launching OncoGrid with: ' + setIds);
+      
+      if (_isLaunchingAnalysis) {
+        return;
+      }
+
+      _isLaunchingAnalysis = true;
+      
+      var payload = {
+        donorSet: _this.selectedForOnco.donor,
+        geneSet: _this.selectedForOnco.gene
+      };
+      
+      var promise = Restangular.one('analysis').post('oncogrid', payload, {}, {'Content-Type': 'application/json'});
+      
+      promise.then(function(data) {
+        if (data.id) {
+          $location.path('analysis/view/oncogrid/' + data.id);
+        }
```

Comment:
Looks like this might fail silently, should probably throw an error

recommend

```js
//...
.then(function (data) {
  if(!data.id) {
    throw new Error('Received invalid response from analysis creation');
  }
  $location.path('analysis/view/oncogrid/' + data.id);
})
//...
```